test/unit/host_test.rb

```diff
@@ -1123,6 +1123,19 @@ def teardown
       assert_kind_of Nic::Managed, host.interfaces.find_by_identifier('eth1')
     end
 
+    test "host can't have more interfaces with same identifier" do
+      host = FactoryGirl.build(:host, :managed)
+      host.primary_interface.identifier = 'eth0'
+      nic = host.interfaces.build(:identifier => 'eth0')
+      refute host.valid?
+      assert_present nic.errors[:identifier]
+      assert_present host.errors[:interfaces]
+      nic.identifier = 'eth1'
+      host.valid?
```

Comment:
use validate, valid? is logically a question, and makes little sense in this context...
can you put this into context by doing
```ruby
assert host.invalid?
```

in rails 4 you can host.validate! which does make sense without assertion, but we're not there yet.